### ISSUE TITLE
Union type is not compared correctly

### ISSUE THREAD

**Issue by Kein** [microsoft/pylance-release#698]
## Environment data
0. VScode 1.47.2
1. Win7x64
2. Python 3.8.5 win64,
3. Pylance language server 2020.12.0

## Expected behaviour

Type "inherited"

## Actual behaviour

Type error reported

## Sample

```python
import codecs
from typing import Union

def parse_string(start: int, end: int, data: Union[bytes, bytearray, memoryview]) -> str:
    tStr = codecs.decode(data[start:end], "latin")
    return tStr

q = b'ABCDEFG'
res = parse_string(0,5, memoryview(q))
print(res)
```
![image](https://user-images.githubusercontent.com/623756/101429509-2a7e9500-3914-11eb-9671-cb844b184849.png)
![image](https://user-images.githubusercontent.com/623756/101429585-5568e900-3914-11eb-8b08-738fdb2c7ab1.png)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.12.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020121-9-december-2020
